tests/test_diffraction_objects.py

```diff
@@ -335,8 +337,8 @@ def test_dump(tmp_path, mocker):
 
 @pytest.mark.parametrize("inputs, expected", tc_params)
 def test_constructor(inputs, expected):
-    actual_do = DiffractionObject(**inputs)
-    diff = DeepDiff(actual_do.__dict__, expected, ignore_order=True, significant_digits=13)
+    actual = DiffractionObject(**inputs).__dict__
+    diff = DeepDiff(actual, expected, ignore_order=True, significant_digits=13, exclude_paths="root['_id']")
```

Comment:
for testing `constructor`, `_id` is ignored since it's hard to dynamically compare its uuid value.

Reply:
is that ok? not sure if there is a better way for this.

Reply:
In the past I have mocked that attribute.  This seems less work.  If we have tests that test the id setting and getting, we can safely ignore it in other tests that are testing other things I think

Reply:
Noted - yes is setting/getting provided below.